### PR TITLE
feat: Add code compatibility with landing on stars.

### PR DESCRIPTION
**Pull request by Codetoil**
Rewrote it to comply more with standards. Please give more feedback. Thanks!
Old PR: #25

### CODE REVIEW

**Comment by Codetoil** [comment]
@marcus8448 New PR for my thingy!

File: src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java [hunk 1]
```diff
@@ -33,7 +32,7 @@
  *
  * @param <C> the type of configuration
  */
-public interface Landable<C extends CelestialBodyConfig> extends SurfaceEnvironment<C> {
+public interface Landable<C extends CelestialBodyConfig> {
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java:35]: Why did you remove the `SurfaceEnvironment` extension? All subtypes of `Landable` have this interface directly applied now (which kind of defeats the purpose)

File: src/main/java/dev/galacticraft/impl/universe/position/config/SatelliteConfig.java [hunk 2]
```diff
@@ -54,8 +56,8 @@ public final class SatelliteConfig implements CelestialBodyConfig {
             LevelStem.CODEC.fieldOf("dimension_options").forGetter(SatelliteConfig::dimensionOptions)
     ).apply(instance, SatelliteConfig::new));
 
-    private final ResourceKey<CelestialBody<?, ?>> parent;
-    private final ResourceKey<Galaxy> galaxy;
+    private final Optional<ResourceKey<CelestialBody<?, ?>>> parent;
```
Comment (marcus8448) [inline src/main/java/dev/galacticraft/impl/universe/position/config/SatelliteConfig.java:59]: Satellites should always have a parent body

**Review by marcus8448 — CHANGES_REQUESTED** [review]
Thanks for taking the time to resubmit this PR. Sadly, landable stars are out-of-scope for Galacticraft. If you wish to have landable stars with Galacticraft, please create an addon - you can still register your own types via `AddonRegistry.CELESTIAL_BODY_TYPE`.

On the other hand, I would agree with having the `Galaxy` be optional for all celestial bodies and allowing the default planet type to not have a direct `parent`. Feel free to submit these changes in a future PR.

(As a note for future PRs, please try to keep curly braces on the same line and use 4 spaces for indentation. Thanks!)

File: src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java (hunk 1 above)
Comment (Codetoil) [reply src/main/java/dev/galacticraft/api/universe/celestialbody/landable/Landable.java:35]: Because things that are not `Landable` have the `SurfaceEnviornment` extension as well. (Decorative planets and Decorative stars)

File: src/main/java/dev/galacticraft/impl/universe/position/config/SatelliteConfig.java (hunk 2 above)
Comment (Codetoil) [reply src/main/java/dev/galacticraft/impl/universe/position/config/SatelliteConfig.java:59]: I was thinking of rouge satellites not attached to any celestial body.

**Comment by Codetoil** [comment]
I'd argue that even though landable stars are outside the scope of the galacticraft itself, It would still be a good idea to add them to the api, just for the fact that they might be implemented in add-ons. For GC4, it was impossible to make stars landable without [patching the code](https://github.com/Codetoil/ToTheStars), so I thought the same would be the case for GC5. It doesn't look like that's correct, but regardless, I plan to also create a similar PR but for GC4. Maybe if it gets added to GC4 you'll consider adding it to GC5?
